tokio/src/sync/oneshot.rs

```diff
@@ -1014,6 +1014,35 @@ impl<T> Receiver<T> {
         self.inner = None;
         result
     }
+
+    /// Blocking receive to call outside of asynchronous contexts.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if called within an asynchronous execution
+    /// context.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::thread;
+    /// use tokio::sync::oneshot;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let (tx, rx) = oneshot::channel::<u8>();
+    ///
+    ///     let sync_code = thread::spawn(move || {
+    ///         assert_eq!(Some(10), rx.blocking_recv());
+    ///     });
+    ///
+    ///     let _ = tx.send(10);
+    ///     sync_code.join().unwrap();
+    /// }
+    /// ```
+    pub fn blocking_recv(self) -> Option<T> {
+        crate::future::block_on(self).ok()
+    }
```

Comment:
This should be using the `RecvError` like `.await` does for consistency.

Reply:
Both [`mpsc::Receiver::blocking_recv`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html#method.blocking_recv) and [`mpsc::UnboundedReceiver::blocking_recv`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.UnboundedReceiver.html#method.blocking_recv) return `Option`. Wouldn't that be inconsistent?